pkg/solana/bindings.go

```diff
@@ -0,0 +1,103 @@
+package solana
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types"
+)
+
+type readBinding interface {
+	GetLatestValue(ctx context.Context, params, returnVal any) error
+	Bind(types.BoundContract) error
+	CreateType(bool) (any, error)
+}
+
+// key is namespace
+type namespaceBindings map[string]methodBindings
+
+// key is method name
+type methodBindings map[string]readBindings
+
+// read bindings is a list of bindings by index
+type readBindings []readBinding
+
+func (b namespaceBindings) AddReadBinding(namespace, methodName string, reader readBinding) {
+	nbs, nbsExists := b[namespace]
+	if !nbsExists {
+		nbs = methodBindings{}
+		b[namespace] = nbs
+	}
+
+	rbs, rbsExists := nbs[methodName]
+	if !rbsExists {
+		rbs = []readBinding{}
+	}
+
+	b[namespace][methodName] = append(rbs, reader)
+}
+
+func (b namespaceBindings) GetReadBindings(namespace, methodName string) ([]readBinding, error) {
+	nbs, nbsExists := b[namespace]
+	if !nbsExists {
+		return nil, fmt.Errorf("%w: no read binding exists for %s", types.ErrInvalidConfig, namespace)
+	}
+
+	rbs, rbsExists := nbs[methodName]
+	if !rbsExists {
+		return nil, fmt.Errorf("%w: no read binding exists for %s and %s", types.ErrInvalidConfig, namespace, methodName)
+	}
+
+	return rbs, nil
+}
+
+func (b namespaceBindings) CreateType(namespace, methodName string, forEncoding bool) (any, error) {
+	bindings, err := b.GetReadBindings(namespace, methodName)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(bindings) == 1 {
+		// get the item type from the binding codec
+		return bindings[0].CreateType(forEncoding)
+	}
+
+	// default to map when multiple bindings exist
+	return &map[string]any{}, nil
```

Comment:
I don't think this will work.  Big ints being decoded to any can cause overflow errors.  If this is so you can read from multiple accounts, I think you need to make a merged struct with the two results.

Reply:
Such as looping through the bindings and creating a struct from all the types returned from each?

Reply:
I used `reflect.StructOf` to create a merged struct.